src/core/search/block_list_test.cc

```diff
@@ -0,0 +1,123 @@
+// Copyright 2023, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "core/search/block_list.h"
+
+#include <absl/container/btree_set.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <set>
+
+#include "base/gtest.h"
+#include "base/logging.h"
+
+namespace dfly::search {
+
+using namespace std;
+
+template <typename C> class BlockListTest : public testing::Test {
```

Comment:
You could keep a list of pointers to all `Make()`-ed lists, add a check in the d'tor to make sure that:
1. All elements are sorted
2. All block sizes are within the specified size
3. No blocks are empty

Reply:
Too much hassle
1. I already check this with tests
2. Added a dcheck
3. same as (2)